app/src/main/java/org/xbmc/kore/ui/generic/SubtitlesOptionsDialog.java

```diff
@@ -0,0 +1,313 @@
+package org.xbmc.kore.ui.generic;
+
+
+import android.app.Activity;
+
+import android.os.Handler;
+import android.support.v4.app.FragmentActivity;
+import android.text.TextUtils;
+import android.widget.Toast;
+
+import org.xbmc.kore.R;
+import org.xbmc.kore.host.HostConnectionObserver;
+import org.xbmc.kore.host.HostInfo;
+import org.xbmc.kore.host.HostManager;
+import org.xbmc.kore.jsonrpc.ApiCallback;
+import org.xbmc.kore.jsonrpc.ApiMethod;
+import org.xbmc.kore.jsonrpc.HostConnection;
+import org.xbmc.kore.jsonrpc.method.Addons;
+import org.xbmc.kore.jsonrpc.method.GUI;
+import org.xbmc.kore.jsonrpc.method.Input;
+import org.xbmc.kore.jsonrpc.method.Player;
+import org.xbmc.kore.jsonrpc.type.ListType;
+import org.xbmc.kore.jsonrpc.type.PlayerType;
+import org.xbmc.kore.utils.LogUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * Dialog that presents a list of options related to subtitles.
+ */
+public class SubtitlesOptionsDialog
+        implements HostConnectionObserver.PlayerEventsObserver,
```

Comment:
Do you assume the user will keep the dialog open while searching and downloading subtitles in Kodi? I would expect them to close the dialog and use the remote (control pad) to manage Kodi.